network/p2p/peerstore/utils.go

```diff
@@ -49,3 +52,18 @@ func PeerInfoFromAddr(addr string) (*peer.AddrInfo, error) {
 	}
 	return info, nil
 }
+
+// PeerInfoFromDomainPort converts a string of the form domain:port to AddrInfo
+func PeerInfoFromDomainPort(domainPort string) (*peer.AddrInfo, error) {
+	parts := strings.Split(domainPort, ":")
+	if len(parts) != 2 {
+		return nil, fmt.Errorf("invalid domain port string %s, found %d colon-separated parts", domainPort, len(parts))
+	}
+	maddr, err := multiaddr.NewMultiaddr(fmt.Sprintf("/dns4/%s/tcp/%s", parts[0], parts[1]))
+	if err != nil {
+		return nil, err
+	}
+	// These will never have peer IDs
+	transport, _ := peer.SplitAddr(maddr)
+	return &peer.AddrInfo{ID: peer.ID(domainPort), Addrs: []multiaddr.Multiaddr{transport}}, nil
```

Comment:
Yeah, I like that we have a unique ID here. If the peer has an ID and we try to connect (using libp2p), there is some code we can write to figure out what it is. 
See https://github.com/libp2p/go-libp2p/pull/1823/files https://github.com/libp2p/go-libp2p/issues/2449 https://github.com/ipfs-shipyard/vole/pull/27/files for more details on that.

I'm slightly worried about how we're going to mix these sorts of peers with peers that have a valid ID. For example, if I'm adding peers to the peerstore via the DHT they won't have addressData, and vice versa if I try to use one of these peers in any libp2p code it will fail to connect. Maybe this is a concern for another PR in which we're actually integrating these two things? Curious on if you have any thoughts about it though.

Reply:
> Yeah, I like that we have a unique ID here. If the peer has an ID and we try to connect (using libp2p), there is some code we can write to figure out what it is.

The conversation is outside of the scope of this PR but I'm not sold that we want to enable this without heavy caveats about it potentially exposing you to MITM attacks. 



> I'm slightly worried about how we're going to mix these sorts of peers with peers that have a valid ID. For example, if I'm adding peers to the peerstore via the DHT they won't have addressData, and vice versa if I try to use one of these peers in any libp2p code it will fail to connect. Maybe this is a concern for another PR in which we're actually integrating these two things? Curious on if you have any thoughts about it though.

The vice-versa should work right? Given a valid p2p multi-addr you can get a valid host/ip:port from it, and we can distribute valid multiaddrs including p2pIDs via DHT for peers that have it enabled.

Reply:
> if I'm adding peers to the peerstore via the DHT they won't have addressData
This PR creates a drop-in replacement for the phonebook. I think the next step is to figure out how we integrate the phonebook with DHT?

Reply:
And to @iansuvak's point on the security. It is noted in one of the PRs
```
// DisablePeerIDCheck disables checking the remote peer ID for a noise connection.
// For outbound connections, this is the equivalent of calling `SecureInbound` with an empty
// peer ID. This is susceptible to MITM attacks since we do not verify the identity of the remote
// peer
```

Reply:
Our current ws network fetches DNS entries without verifying any pre-defined public keys as well. The existing [identity implementation](https://github.com/algorand/go-algorand/blob/master/network/netidentity.go) (which is only enabled if you alter the default empty PublicAddress in your config) is a way for two peers to tell each other "this is my public key" and for those messages to be signed so that you know the peer holds the corresponding private key. But once again those keys are arbitrary--i.e. any random peer with a key will do.
The only sort of cryptographic security we provide around the identity of relays is dnssec which would still be in place for wsNetwork relays. So in the case of disabling the peerID check for the wsNetwork it's equivalent behavior afaict. 

As long as we're using this in the same way as [kubo](https://github.com/libp2p/specs/issues/458#issuecomment-1255951857) it's fine, i.e. you just want some random peer...
> this feature has no security implications as you're just asking kubo to form a connection to some random peer.

@AlgoAxel feel free to correct me on the bits around the conn dedup that I've gotten wrong.

Reply:
I agree that it's the same as the current behavior but if we have the mechanisms to  share and advertise public keys I think we should prefer that in all cases to connecting blindly. Particularly if we end up storing non-relay peers into the peerstore backed phonebook.